plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -331,4 +349,30 @@ private static void flushPreferences(IEclipsePreferences preferences) {
       this.shell = shell;
     }
   }
+
+  @VisibleForTesting
+  String jsonEncode(PingEvent event) {
+    Map<String, Object> root = new HashMap<>();
+    Map<String, Object> clientInfo = new HashMap<>();
+    Map<String, String> desktopClientInfo = new HashMap<>();
+    desktopClientInfo.put("os", System.getProperty("os.name"));
+    clientInfo.put("client_type", "DESKTOP");
+    clientInfo.put("desktop_client_info", desktopClientInfo);
+    root.put("log_source_name", "CONCORD");
+    root.put("zwieback_cookie", uuid);
+    root.put("request_time_ms", System.currentTimeMillis());
+    root.put("client_info", clientInfo);
+    
+    List<Map<String, Object>> logEvents = new ArrayList<>(1);
+    Map<String, Object> logEvent = new HashMap<>();
+    logEvent.put("event_time_ms", System.currentTimeMillis());
+    logEvent.put("sequence_position", sequencePosition++);
+    
+    logEvent.put("source_extension_json", 
+        "{\"console_type\":\"CLOUDCODE_INTELLIJ\",\"event_name\":\"some_event_name\",\"event_metadata\":[{\"some_key\":\"some_value\"}]}");
```

Comment:
Probably this is to be expanded later so that this Json part is constructed through Java objects like the others rather than embedding the raw Json string?

Reply:
Never mind. Just saw you did it.